backend/poetry.lock

```diff
@@ -1,4 +1,4 @@
-# This file is automatically @generated by Poetry 1.6.1 and should not be changed by hand.
```

Comment:
I think you are using a different version of Poetry locally than what is used by the build / production.

RUN pip install poetry==1.6.1